File: mall_web/src/main/resources/static/js/service/brandService.js
app.service("brandService",function ($http) {
    //从异步请求中读取数据到页面
    this.findAll=function () {
        return $http.get("../brand/findAll");
    };

    this.findPage=function (page, size) {
        return $http.get('../brand/findPage?page=' + page + '&size=' + size);
    };
    
    this.findById=function (id) {
        return $http.get("../brand/findById?id="+id);
    };
    
    this.add=function (entity) {
        return $http.post("../brand/add", entity);
    };

    this.update=function (entity) {
        return $http.post("../brand/updateBrand",entity)
    };

    this.delete=function (selectIds) {
        return $http("../brand/deleteBrand?ids=" + selectIds);
    }
});

File: mall_web/src/main/java/com/shop/mall/mall_web/service/BrandService.java
package com.shop.mall.mall_web.service;

import com.shop.mall.mall_web.pojo.TbBrand;
import com.shop.mall.mall_web.pojo.entity.PageResult;

import java.util.List;

public interface BrandService {
    /**
     * 获取品牌表中的所有数据
     */
    List<TbBrand> findAll();

    /**
     * @param page
     * @param size
     * @return
     * 分页查询品牌列表
     */
    PageResult<TbBrand> findPage(Integer page,Integer size,TbBrand brand);

    /**
     * @param tbBrand
     * @return
     * 插入品牌信息
     */
    Boolean insertBrand(TbBrand tbBrand);

    /**
     * @param tbBrand
     * @return
     * 更新品牌信息
     */
    Boolean updateBrand(TbBrand tbBrand);

    /**
     * @param id
     * @return
     * 根据id查询品牌信息
     */
    TbBrand findById(Long id);

    /**
     * @param ids
     * @return
     * 根据id删除品牌信息
     */
    Boolean deleteBrand(Long[] ids);
}
